nexus_constructor/model/transformation.py

```diff
@@ -46,9 +51,21 @@ def vector(self, new_vector: QVector3D):
 
     @property
     def ui_value(self) -> float:
-        if isinstance(self.values, (float, int)):
-            return float(self.values)
-        return float(self.get_attribute_value(CommonAttrs.UI_VALUE))
+
+        ui_value = self.get_attribute_value(CommonAttrs.UI_VALUE)
```

Comment:
I think this should be checking the magnitude of the dataset value first, rather than the UI value attribute, and seeing if it is an int/double. Then if it's anything else it should revert to using the `ui_value` attribute.